src/dimindexing.jl

```diff
@@ -1,14 +1,38 @@
 function dim_to_indices(arr::IndexedDimsArray{T, N}, inds::Tuple{Vararg{Any, N}}) where {T, N}
+    @show arr.indexes
+    @show inds
     return map(unwrap_index, arr.indexes, inds)
 end
 
-unwrap_index(acc::AcceleratedVector, ind) = findall(in(ind), acc)
-unwrap_index(acc::AcceleratedVector, ::Colon) = Colon()
-unwrap_index(acc::AcceleratedVector, ind::BypassIndex) = ind.index
+function unwrap_index(acc::AcceleratedVector{T, <:AbstractVector{T}, <:AcceleratedArrays.AbstractUniqueIndex}, ind) where T
+    return _unwrap_unique_index(acc, ind)
+end
+_unwrap_unique_index(acc::AcceleratedVector, ind) = findfirst(isequal(ind), acc)
+_unwrap_unique_index(acc::AcceleratedVector, ind::Union{AbstractArray, Colon, BypassIndex}) = _unwrap_index(acc, ind)
+unwrap_index(acc::AcceleratedVector, ind) = _unwrap_index(acc, ind)
+_unwrap_index(acc::AcceleratedVector, ind) = findfirst(isequal(ind), acc)
```

Comment:
```suggestion
# Note: it is *not* part of the API that this will use `findfirst` only yhast it will find one
_unwrap_index(acc::AcceleratedVector, ind) = findfirst(isequal(ind), acc)
```